BUILDING.md

```diff
@@ -0,0 +1,276 @@
+on the transitive dependencies of those projects.
+
+You will need to install these required dependencies before building
+the Viam C++ SDK, preferably by way of your system package manager:
+
+- Debian: `apt-get install cmake build-essential libboost-all-dev libgrpc++-dev libprotobuf-dev`
+- MacOS with Homebrew: `brew install cmake boost grpc protobuf`
+
+There are also several optional or conditionally required dependencies:
+
+- Debian: `apt-get install pkg-config ninja protobuf-compiler-grpc`
+- MacOS with Homebrew: `brew install pkg-config ninja buf`
+
+Note that Debian Bullseye does not ship a recent enough version of
+CMake by default, but it can be obtained from the Bullseye Backports
+repository:
+
+``` shell
+apt-add-repository 'deb http://deb.debian.org/debian bullseye-backports main'
+apt-get update
+apt-get install -t bullseye-backports cmake
+```
+
+You can of course build and install any or all of these requirements
+manually rather than via your package manager, though this is likely
+to make it more difficult to correctly configure the build of the Viam
+C++ SDK.
+
+## Obtaining the `viam-cpp-sdk` Source
+
+Just clone https://github.com/viamrobotics/viam-cpp-sdk into a
+directory of your choice and checkout the branch or tag of interest.
+
+## Invoking CMake to Generate a Build System
+
+This document doesn't intend to provide a complete guide for how CMake
+is used, as that is a complex topic. Please consult the [CMake
+documentation](https://cmake.org/cmake/help/latest/index.html) for
+specific details on how CMake works and how to invoke the tool. The
+guidance here is intended to cover the common case builds of the
+`viam-cpp-sdk`.
+
+CMake organizes out-of-tree builds. The first step is to create a
+scratch directory where you want to keep the working state of your
+build. A convention is to name this directory `build` and to create it
+under the root of the project. The `viam-cpp-sdk` top-level
+`.gitignore` file already ignores such a directory, so we recommend
+starting there:
+
+``` shell
+git clone https://github.com/viamrobotics/viam-cpp-sdk
+cd viam-cpp-sdk
+mkdir build
+cd build
+```
+
+CMake isn't itself a build system, it is a build system generator. Our
+next step is to generate the build system. In these examples we use
+Ninja, but you can also use Makefiles or [many other build
+systems](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html).
+
+When invoking CMake, you pass the path to the root `CMakeLists.txt`
+file for the project you wish to build. If you have opted to use an
+in-tree `build` directory per the above instructions, this is as
+simple as:
+
+``` shell
+cmake .. -G Ninja
+```
+
+If all went well, you should now have a file named `build.ninja` in
+the `build` directory created above. Note that `git status` will not
+show this, since the `build` directory is ignored.
+
+## Building and Installing the SDK
+
+To compile all targets in the SDK:
+
+``` shell
+ninja all
```

Comment:
maybe mentioning needing to run this twice?

Reply:
I'd rather see if I can figure out what's wrong and push a little fix. Let me poke at it a bit.

Reply:
I think this is fixed in latest. @dannenberg can you give it a try?